adaboost/cuda/core/operations.hpp

```diff
@@ -53,6 +55,13 @@ namespace adaboost
             void multiply_gpu(const MatrixGPU<data_type_matrix>& mat1,
             const MatrixGPU<data_type_matrix>& mat2,
             MatrixGPU<data_type_matrix>& result);
+         
+            template <class data_type_vec, class data_type_ret>
+            void Argmax(
+            func_t<data_type_vec,data_type_ret> p_func,
+            const VectorGPU<data_type_vec>& vec,
+            data_type_vec& result,
+            unsigned int block_size);
```

Comment:
Since, `block_size` is `unsigned`, we should check only for the case `block_size=0` and throw error if it is `0`.

Reply:
Done